predicators/ground_truth_models/cover/nsrts.py

```diff
@@ -316,3 +316,160 @@ def place_on_table_sampler(state: State, goal: Set[GroundAtom],
+
+        pick_from_smooth_nsrt = NSRT("PickFromSmooth", parameters,
+                                     preconditions, add_effects,
+                                     delete_effects, set(), option,
+                                     option_vars, pick_sampler)
+        nsrts.add(pick_from_smooth_nsrt)
+
+        # Pick from bumpy region
+        parameters = [block]
+        preconditions = {
+            LiftedAtom(HandEmpty, []),
+            LiftedAtom(InBumpyRegion, [block])
+        }
+        add_effects = {
+            LiftedAtom(Holding, [block]),
+        }
+        delete_effects = {
+            LiftedAtom(HandEmpty, []),
+            LiftedAtom(InBumpyRegion, [block])
+        }
+        option = PickFromBumpy
+        option_vars = parameters
+
+        pick_from_bumpy_nsrt = NSRT("PickFromBumpy", parameters,
+                                    preconditions, add_effects, delete_effects,
+                                    set(), option, option_vars, pick_sampler)
+        nsrts.add(pick_from_bumpy_nsrt)
+
+        # Place on target
+        parameters = [block, target]
+        preconditions = {
+            LiftedAtom(Holding, [block]),
+            LiftedAtom(Clear, [target]),
+        }
+        add_effects = {
+            LiftedAtom(HandEmpty, []),
+            LiftedAtom(InSmoothRegion, [block]),
+            LiftedAtom(Covers, [block, target]),
+        }
+        delete_effects = {
+            LiftedAtom(Holding, [block]),
+            LiftedAtom(Clear, [target])
+        }
+        option = PlaceOnTarget
+        option_vars = parameters
+
+        def place_on_target_sampler(state: State, goal: Set[GroundAtom],
+                                    rng: np.random.Generator,
+                                    objs: Sequence[Object]) -> Array:
+            del goal  # unused
+            _, t = objs
+            assert t.is_instance(target_type)
+            center = float(state.get(t, "pose"))
+            if CFG.bumpy_cover_right_targets:
+                center += 3 * state.get(t, "width") / 4
+            lb = center - state.get(t, "width") / 2
+            ub = center + state.get(t, "width") / 2
+            lb = max(lb, 0.0)
+            ub = min(ub, 1.0)
+            return np.array(rng.uniform(lb, ub, size=(1, )), dtype=np.float32)
+
+        place_on_target_nsrt = NSRT("PlaceOnTarget", parameters,
+                                    preconditions, add_effects, delete_effects,
+                                    set(), option, option_vars,
+                                    place_on_target_sampler)
+        nsrts.add(place_on_target_nsrt)
+
+        # Place in bumpy region
+        parameters = [block]
+        preconditions = {LiftedAtom(Holding, [block])}
+        add_effects = {
+            LiftedAtom(HandEmpty, []),
+            LiftedAtom(InBumpyRegion, [block])
+        }
+        delete_effects = {LiftedAtom(Holding, [block])}
+        option = PlaceOnBumpy
+        option_vars = parameters
```

Comment:
Can we add a comment here just re-affirming the fact that the targets are never in bumpy regions?